### PR TITLE
fix(side-panel): refactor to match latest ibm-products

### PR DESCRIPTION
**Pull request by lee-chase**
### Related Ticket(s)

https://github.com/carbon-design-system/ibm-products/pull/4141

### Description

Updates structure of SidePanel following restructure of @carbon/ibm-products SidePanel. This simplifies the HTML structure and title scroll animation.

### Changelog

**Changed**

- packages/carbon-web-components/src/components/side-panel/side-panel.scss
- packages/carbon-web-components/src/components/side-panel/side-panel.ts

### CODE REVIEW

File: packages/carbon-web-components/src/components/side-panel/side-panel.scss [hunk 1]
```diff
@@ -327,6 +236,13 @@ $block-class-action-set: #{$pkg-prefix}--action-set;
 
       &[actions-multiple='triple'] {
         --flex-direction: column;
+
+        /* stylelint-disable-next-line selector-type-no-unknown */
+        ::slotted(cds-button) {
```
Comment (ariellalgilmore) [inline packages/carbon-web-components/src/components/side-panel/side-panel.scss:241]:
```suggestion
        ::slotted(`#{$prefix}-button`) {
```

File: packages/carbon-web-components/src/components/side-panel/side-panel.scss [hunk 2]
```diff
@@ -335,15 +251,15 @@ $block-class-action-set: #{$pkg-prefix}--action-set;
       --flex-direction: column;
 
       /* stylelint-disable-next-line selector-type-no-unknown */
-      &[actions-multiple] ::slotted(cds-button) {
-        flex: 0 0 100%;
+      ::slotted(cds-button) {
```
Comment (ariellalgilmore) [inline packages/carbon-web-components/src/components/side-panel/side-panel.scss:254]:
```suggestion
      ::slotted(`#{$prefix}-button`) {
```

**Review by ariellalgilmore — APPROVED** [review]
Looks good! just minor adjustments

**Comment by lee-chase** [comment]
@ariellalgilmore corrections made, there were a few 'cds-' now replaced.

**Pull request merged**